internal/impl/shell.go

```diff
@@ -411,3 +411,21 @@ func filterPathList(pathList string, keep func(string) bool) string {
 	}
 	return strings.Join(filtered, string(filepath.ListSeparator))
 }
+
+func findNixInPATH(env map[string]string) (string, error) {
+	defaultSingleUserNixBin := fmt.Sprintf("%s/.nix-profile/bin", env["HOME"])
+	defaultMultiUserNixBin := "/nix/var/nix/profiles/default/bin"
+	xdgNixBin := xdg.StateSubpath("/nix/profile/bin")
+	pathElements := strings.Split(env["PATH"], ":")
+	debug.Log("path elements: %v", pathElements)
+	for _, el := range pathElements {
+		if el == xdgNixBin ||
+			el == defaultSingleUserNixBin ||
+			el == defaultMultiUserNixBin {
+			return el, nil
+		}
```

Comment:
I sort of think we should include all that are specified instead of just the first. It's not a big deal, but it would be more defensive code.

Reply:
@mike isn't your suggestion contradicting your comment [here](https://github.com/jetpack-io/devbox/pull/1143#discussion_r1227248076)?